include/pybind11/cast.h

```diff
@@ -42,13 +42,15 @@ using make_caster = type_caster<intrinsic_t<type>>;
 // Shortcut for calling a caster's `cast_op_type` cast operator for casting a type_caster to a T
 template <typename T>
 typename make_caster<T>::template cast_op_type<T> cast_op(make_caster<T> &caster) {
-    return caster.operator typename make_caster<T>::template cast_op_type<T>();
+    using result_t = typename make_caster<T>::template cast_op_type<T>;
```

Comment:
Could you please add a comment at the end of the line, e.g.

```
// See PR #4893.
```

Also below.

In case someone later is considering inlining `result_t` again, they can easily know why it's here.